src/solvers/dgsem_unstructured/sort_boundary_conditions.jl

```diff
@@ -41,10 +41,29 @@ function initialize!(boundary_types_container::UnstructuredSortedBoundaryTypes{N
 
   unique_names = unique(cache.boundaries.name)
 
-  # TODO: This needs to be handled differently for the `ParallelP4estMesh` since the boundaries
-  # are distributed and thus unique_names only contains the names of boundaries on the local process
-  # See https://github.com/trixi-framework/Trixi.jl/issues/1047
-  if !mpi_isparallel()
+  if mpi_isparallel()
+    # Exchange of boundaries names
+    unique_names_byte=Vector{UInt8}(join(unique_names, "\0"))
+    if(length(unique_names)!=0)
+      push!(unique_names_byte,0)
+    end
```

Comment:
Why do you need this `if`? Couldn't you just do
```suggestion
    push!(unique_names_byte, 0)
```

Reply:
It was needed in case, when process sends empty `unique_names`. Because then `Symbol("")` will be in the gathered array. And will cause error, when root process will check whether all boundaries have boundary condition or not. But with your suggestions below, to do not overwrite `unique_names` and to use `split(...;keepempty=false)`, this `if` is unnecessary.